Client/mods/deathmatch/logic/CClientCamera.cpp

```diff
@@ -644,3 +648,65 @@ void CClientCamera::ResetShakeCamera() noexcept
 {
     m_pCamera->ResetShakeCamera();
 }
+
+bool CClientCamera::IsInCameraTransition() const
+{
+    if (!m_pCamera)
+        return false;
+    
+    return m_pCamera->GetTransitionState() != 0;
+}
+
+CMatrix CClientCamera::GetInterpolatedCameraMatrix() const
+{
+    if (!m_pCamera)
+        return CMatrix();
+    
+    CCameraSAInterface* pCameraInterface = static_cast<CCameraSA*>(m_pCamera)->GetInterface();
+    if (!pCameraInterface)
+        return CMatrix();
+    
+    CVector source = pCameraInterface->SourceDuringInter;
+    CVector target = pCameraInterface->TargetDuringInter;
+    CVector up = pCameraInterface->UpDuringInter;
+    
+    CMatrix matrix;
+    CVector forward = target - source;
+    if (forward.Length() < FLOAT_EPSILON)
+        forward = CVector(0.0f, 1.0f, 0.0f);
+    else
+        forward.Normalize();
+    
+    CVector right = CVector(forward.fY, -forward.fX, 0.0f);
+    if (right.Length() < FLOAT_EPSILON)
+        right = CVector(1.0f, 0.0f, 0.0f);
+    else
+        right.Normalize();
+    
+    CVector correctedUp = right;
+    correctedUp.CrossProduct(&forward);
+    correctedUp.Normalize();
+    
+    matrix.vPos = source;
+    matrix.vFront = forward;
+    matrix.vRight = -right;
+    matrix.vUp = correctedUp;
+    matrix.OrthoNormalize(CMatrix::AXIS_FRONT, CMatrix::AXIS_UP);
+    
+    return matrix;
+}
+
+float CClientCamera::GetAccurateFOV() const
+{
+    if (!m_pCamera)
+        return 70.0f;
+    
+    CCameraSAInterface* pCameraInterface = static_cast<CCameraSA*>(m_pCamera)->GetInterface();
+    if (!pCameraInterface)
+        return 70.0f;
+    
+    if (IsInCameraTransition())
+        return pCameraInterface->FOVDuringInter;
+    
+    return *(float*)0x8D5038;
```

Comment:
Always document such addresses. Either define them as `constexpr std::uintptr_t` or add a comment. We don’t want such magic values.

https://github.com/multitheftauto/mtasa-docs/blob/main/mtasa-blue/CODING_GUIDELINES.md#avoid-magic-numbers.